pull_request_template.md

```diff
@@ -0,0 +1,35 @@
+# Overview:
+Give a brief overview of the issue you are solving. Succintly explain the GitHub issue you are addressin and the underlying problem of the ticket.
+
+# Addresses:
+Add a link to the issue on Github board
+
+# Type of Fix:
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] Documentation Change
+
+# Solution:
+Outline your solution to the previously described issue and underlying cause. This section should include a brief description of your proposed solution and how it addresses the cause of the ticket.
+
+# Adds:
+Include a bulleted list or check box list of the implemented changes in brief, as well as the addition of supplementary materials(unit tests, integration tests, etc..)
+
+# Validation:
+Describe how you have validated that your solution addresses the root cause of the ticket. What have you done to ensure that your addition is bug free and works as expected. Please provide specific instructions so we can reproduce and list any relevant details about your configuration.
+- Screenshots
+- Unit Tests
+- Script to reproduce error
+- Configuration details
+
+# Checklist
+ - [ ] My code follows Google style guide
+ - [ ] My code is unit tested where appropriate and does not decrease test coverage
+ - [ ] I have performed a self review of my own code
+ - [ ] I have commented my code, specifically adding doc strings to any added functions
+ - [ ] I have updated the documentation of the repository where appropriate
+
```

Comment:
Please add a bullet for checking that commits follow our guidelines, and the header commit includes the issue number.

Reply:
added

Reply:
Additional possible checklist items:

`- [ ] My Pull Request has the latest AllenSDK release candidate branch rc/x.y.z as its merge target`
`- [ ] My code passes all AllenSDK tests`

Reply:
I agree with the first bullet, added.
They should be seeing the test status on the PR but another reminder is a good idea too!

Reply:
what about non-docstring style?

Reply:
you mean like in-line comments? or external documentation? I don't think we set a standard for inline comments that I'm aware of, but I could be wrong.

Reply:
By specifying a docstring format you've covered one kind of style question. There are more! For instance:
- when, if ever, should I use lambdas/comprehensions/generators/metaclasses/...?
- how should I indent when breaking up long lines?

Some rules of this kind are just for consistent appearance. Others matter for the actual design (like: "metaclasses, yea or nay"). If we have a style guide (I thought we had decided on [Google's](https://google.github.io/styleguide/pyguide.html) at some point) this would be a good place to mention it.